src/config.ts

```diff
@@ -0,0 +1,38 @@
+import type { MergeableRecord} from './util/objects'
+import {deepMergeObject} from './util/objects'
+import path from 'path'
+import fs from 'fs'
+import {log} from './util/log'
+
+export interface FlowrConfigOptions extends MergeableRecord {
+	ignoreSourceCalls: boolean
+}
+
+export const defaultConfigFile = 'flowr.json'
+export const defaultConfigOptions: FlowrConfigOptions = {
+	ignoreSourceCalls: false
+}
+export const config = parseConfigOptions(process.cwd())
+
+function parseConfigOptions(workingDirectory: string, configFile = defaultConfigFile): FlowrConfigOptions {
+	let searchPath = path.resolve(workingDirectory)
+	do{
+		const configPath = path.join(searchPath, configFile)
+		if(fs.existsSync(configPath)) {
+			try {
+				const read = fs.readFileSync(configPath,{encoding: 'utf-8'})
+				// assign default values to all config options except for the specified ones
+				const ret = deepMergeObject(defaultConfigOptions, JSON.parse(read) as FlowrConfigOptions)
```

Comment:
I think we should safe guard here with a strict json schema (as used for the network messages). Just to be safe(r) :D